src/main/java/Problem3/Problem3.java

```diff
@@ -6,6 +6,15 @@
 
 public class Problem3 {
     public static void printListInReverse(ListNode list, PrintStream print) {
-        // homework
+        if(list != null && list.next != null){
+            printListInReverse(list.next,print);
+            print.print(list.val + " ");
+            return;
+        }
+        if(list != null && list.next == null){
+            print.print(list.val + " ");
+            return;
+        }
```

Comment:
a bit over thinking here.

```java
    public static void printListInReverse(ListNode list, PrintStream print) {
        if (list == null) {
            return;
        }
        printListInReverse(list.next, print);
        print.println(list.val);
    }
```
this would just do.